src/util/performancetimer.h

```diff
@@ -1,75 +1,39 @@
-** License version 2.1 as published by the Free Software Foundation and
-** appearing in the file LICENSE.LGPL included in the packaging of this
-** file. Please review the following information to ensure the GNU Lesser
-** General Public License version 2.1 requirements will be met:
-** http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
-**
-** In addition, as a special exception, Nokia gives you certain additional
-** rights. These rights are described in the Nokia Qt LGPL Exception
-** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
-**
-** GNU General Public License Usage
-** Alternatively, this file may be used under the terms of the GNU General
-** Public License version 3.0 as published by the Free Software Foundation
-** and appearing in the file LICENSE.GPL included in the packaging of this
-** file. Please review the following information to ensure the GNU General
-** Public License version 3.0 requirements will be met:
-** http://www.gnu.org/copyleft/gpl.html.
-**
-** Other Usage
-** Alternatively, this file may be used in accordance with the terms and
-** conditions contained in a signed written agreement between you and Nokia.
-**
-**
-**
-**
-**
-** $QT_END_LICENSE$
-**
-****************************************************************************/
-
 #pragma once
+#include <chrono>
 
-//
-// This is a fork of QPerformanceTimer just without the Q prefix
-// To fix interface changes issues in different QT versions
-// Added restart() function.
-// Returns time in nanosecond resolution.
-//
+#include "util/duration.h"
 
-#include <QtCore/qglobal.h>
+class PerformanceTimer {
+  public:
+    // TODO: make this configurable via a template parameter?
+    // note that the resolution of std::chrono::steady_clock is not guaranteed
+    // to be high resolution, but it is guaranteed to be monotonic.
+    // However, on all major platforms, it is high resolution enough.
+    using ClockT = std::chrono::steady_clock;
+    PerformanceTimer()
+            : m_startTime(ClockT::duration(0)){};
 
-#include "util/duration.h"
+    void start() {
+        m_startTime = ClockT::now();
+    };
 
-class PerformanceTimer
-{
-public:
-    PerformanceTimer() {
-      t1 = 0;
-#if defined(Q_OS_UNIX) && !defined(Q_OS_MAC)
-      t2 = 0;
-#endif
+    mixxx::Duration elapsed() const {
+        return mixxx::Duration::fromStdDuration(ClockT::now() - m_startTime);
+    };
+    mixxx::Duration restart() {
+        const ClockT::time_point now = ClockT::now();
+        const auto dur = mixxx::Duration::fromStdDuration(now - m_startTime);
+        m_startTime = now;
+        return dur;
     };
 
-    void start();
-    mixxx::Duration elapsed() const;
-    mixxx::Duration restart();
-    mixxx::Duration difference(const PerformanceTimer& timer) const;
+    mixxx::Duration difference(const PerformanceTimer& timer) const {
+        return mixxx::Duration::fromStdDuration(m_startTime - timer.m_startTime);
+    };
```

Comment:
What is your plan with this function? I think getting rid of it was the source of our long discussion. Now it is still there?  

Right now `std::chrono::time_point()` is a private implementation detail which can be easily replaced with the native faster representation.

Reply:
> What is your plan with this function? I think getting rid of it was the source of our long discussion. Now it is still there?

Yes, I want to get rid of and I have a commit flying around somewhere that does that but I didn't want to blow up the PR more than it already is. Step 1 was to simplify the internals, step 2 (separate PR) would be to change the API from `mixxx::Duration` to `chrono::duration`, as well as getting rid of `PerformanceTimer::difference`.

Reply:
OK